### ISSUE TITLE
Issue #63: Time-Aware Train/Test Split (Fix Temporal Leakage)

### ISSUE THREAD

**Issue by wildwasser** [wildwasser/sneaker#63]
## Problem

**CRITICAL: Current 77.64% test accuracy is artificially inflated by temporal leakage**

Random `train_test_split()` allows model to see future (2025) data during training, then "predict" past (2024) data in test set. This creates unrealistic accuracy that doesn't translate to real trading.

**Evidence:**
- 77.64% test accuracy with random split (Issue #62)
- Backtest FAILED: Sharpe -0.27, Win Rate 50%, Return -11.93%
- Model had 4:1 BUY bias in DOWN market (distribution mismatch)

## Root Cause: Temporal Leakage

**Random Split (CURRENT - BROKEN):**
```
Train: [2021-Jan, 2023-Mar, 2024-Nov, 2025-Apr, ...]  ← Mixed timeline
Test:  [2022-Feb, 2024-Jan, 2025-Feb, 2023-Dec, ...]  ← Mixed timeline
```

Model sees 2025 data during training, then "predicts" 2024 in test = cheating!

**Time-Aware Split (FIX - NEEDED):**
```
Train: [2021-Jan → 2024-Oct]  ← First 90% chronologically
Test:  [2024-Nov → 2025-Apr]  ← Last 10% chronologically
```

Model trained on past, tested on future = REAL validation.

## Solution

**Phase 1: ✅ COMPLETE - Infrastructure**
- Modified `experiments/data_loader.py` to add `time_aware_split=True` parameter
- Sorts data by timestamp, splits chronologically (first 90% train, last 10% test)
- Logs train/test period dates to verify no overlap

**Phase 2: ✅ COMPLETE - Training Script**
- Created `scripts/15d_train_binary_timeaware.py`
- Uses `time_aware_split=True` to prevent temporal leakage
- Trains on natural distribution (no sample weighting to avoid oversignaling)
- Same LightGBM parameters as baseline for fair comparison

**Phase 3: ⏳ PENDING - Training & Validation**
- Train on full 1.38M samples with time-aware split
- Expected: Test accuracy will DROP from 77.64% to ~60-65% (realistic)
- Backtest on ARBUSDT 256h to validate real trading performance
- Pass criteria: Sharpe ≥0.5, Win Rate ≥52%

## Expected Outcome

**Realistic Scenario:**
- Test accuracy: 60-65% (down from 77.64%, but REAL)
- Backtest: Sharpe ≥0.5, Win Rate ≥52% (actual predictive power)
- **Outcome:** Features have genuine predictive power → proceed with optimization

**Pessimistic Scenario:**
- Test accuracy: 50-55% (barely better than random)
- Backtest: Sharpe <0.2, Win Rate ~50% (no edge)
- **Outcome:** Lagging indicators can't predict future → need leading indicators or abandon

## Files

**Modified:**
- `experiments/data_loader.py` - Added time-aware split logic

**Created:**
- `scripts/15d_train_binary_timeaware.py` - Training script with time-aware split

**To Create:**
- `proof/issue-63/` - Training results and backtest comparison

## Timeline

1. **Run training:** `scripts/15d_train_binary_timeaware.py` (~5 min)
2. **Backtest validation:** ARBUSDT 256h (~2 min)
3. **Document results:** Compare to random-split baseline
4. **Decision:** Proceed with optimization OR pivot to leading indicators

## References

- Issue #62: Full data training (77.64% with random split)
- Issue #60: Threshold optimization failure (all thresholds failed backtest)
- Issue #61: Regularization failure (accuracy dropped to 57%)

**Comment by starlet219** [comment]
Maybe this update helps.
[f363c51d2a32.zip](https://github.com/user-attachments/files/23697942/f363c51d2a32.zip)
